package/config.js

```diff
@@ -20,6 +20,14 @@ if (isArray(app.extensions) && app.extensions.length) delete defaults.extensions
 
 const config = deepMerge(defaults, app)
 config.outputPath = resolve('public', config.public_output_path)
-config.publicPath = `/${config.public_output_path}/`.replace(/([^:]\/)\/+/g, '$1')
+config.publicPath = `/${config.public_output_path}/`
+
+// Add prefix to publicPath.
+if (process.env.RAILS_RELATIVE_URL_ROOT) {
+  config.publicPath = `/${process.env.RAILS_RELATIVE_URL_ROOT}${config.publicPath}`
```

Comment:
Could you add some test for this, please?

Reply:
I thought about that, but wasn't sure which test file to put them in? I was thinking of using the config test file, but wasn't sure because I didn't see anything in there for `publicPath`.

Reply:
Ahh yes, but could you please add one :)

```js
/* global test expect, describe */

const { chdirTestApp, chdirCwd } = require('../utils/helpers')

chdirTestApp()

describe('Config', () => {
  beforeEach(() => jest.resetModules())
  afterAll(chdirCwd)

  test('public path', () => {
    delete process.env.RAILS_RELATIVE_URL_ROOT
    const config = require('../config')
    expect(config.publicPath).toEqual('/packs/')
  })

  test('public path with relative root', () => {
    process.env.RAILS_RELATIVE_URL_ROOT = '/foo'
    const config = require('../config')
    expect(config.publicPath).toEqual('/foo/packs/')
  })

  test('should return extensions as listed in app config', () => {
    const config = require('../config')
    expect(config.extensions).toEqual([
      '.js',
      '.sass',
      '.scss',
      '.css',
      '.module.sass',
      '.module.scss',
      '.module.css',
      '.png',
      '.svg',
      '.gif',
      '.jpeg',
      '.jpg'
    ])
  })
})
```

Reply:
Will do. Thanks for the boilerplate.

Reply:
I guess we need to set NODE_ENV in the test for consistent results: 

```js
 test('public path', () => {
    process.env.RAILS_ENV = 'development'
    delete process.env.RAILS_RELATIVE_URL_ROOT
    const config = require('../config')
    expect(config.publicPath).toEqual('/packs/')
  })

  test('public path with relative root', () => {
    process.env.RAILS_ENV = 'development'
    process.env.RAILS_RELATIVE_URL_ROOT = '/foo'
    const config = require('../config')
    expect(config.publicPath).toEqual('/foo/packs/')
  })
```

Reply:
I assume you meant `NODE_ENV` in your examples.

Reply:
I assume you meant `NODE_ENV` in your examples.

Reply:
Nope, you meant `RAILS_ENV`.

Reply:
oh yes! I guess you figured it out 😉